tasm-lib/src/arithmetic/u128/overflowing_add.rs

```diff
@@ -0,0 +1,289 @@
+            let ab0 = to_u128(a0, b0, c0, d0);
+
+            // second element on stack
+            let a1: u32 = stack.pop().unwrap().try_into().unwrap();
+            let b1: u32 = stack.pop().unwrap().try_into().unwrap();
+            let c1: u32 = stack.pop().unwrap().try_into().unwrap();
+            let d1: u32 = stack.pop().unwrap().try_into().unwrap();
+            let ab1 = to_u128(a1, b1, c1, d1);
+
+            let (sum, overflow) = ab0.overflowing_add(ab1);
+
+            let mut res = sum.encode();
+            for _ in 0..res.len() {
+                stack.push(res.pop().unwrap());
+            }
+
+            stack.push(bfe!(overflow as u64));
+        }
+
+        fn pseudorandom_initial_state(
+            &self,
+            seed: [u8; 32],
+            _bench_case: Option<BenchmarkCase>,
+        ) -> Vec<BFieldElement> {
+            let mut rng = StdRng::from_seed(seed);
+            let lhs: u128 = rng.gen();
+            let rhs: u128 = rng.gen();
+
+            self.setup_init_stack(lhs, rhs)
+        }
+
+        fn corner_case_initial_states(&self) -> Vec<Vec<BFieldElement>> {
+            let some_none_zero_value = (1u128 << 97) + (1u128 << 65) + u64::MAX as u128 - 456456;
+            let rhs_is_zero = self.setup_init_stack(some_none_zero_value, 0);
+            let lhs_is_zero = self.setup_init_stack(0, some_none_zero_value);
+            let rhs_is_zero_lhs_max = self.setup_init_stack(u128::MAX, 0);
+            let lhs_is_zero_rhs_max = self.setup_init_stack(0, u128::MAX);
+            let both_are_2_to_127 = self.setup_init_stack(1u128 << 127, 1u128 << 127);
+            let two_to_127_var0 = self.setup_init_stack((1u128 << 127) - 1, (1u128 << 127) - 1);
+            let two_to_127_var1 = self.setup_init_stack((1u128 << 127) - 1, 1u128 << 127);
+            let two_to_127_var2 = self.setup_init_stack((1u128 << 127) - 1, (1u128 << 127) + 1);
+            let two_to_127_var3 = self.setup_init_stack(1u128 << 127, (1u128 << 127) - 1);
+            let two_to_127_var4 = self.setup_init_stack(1u128 << 127, (1u128 << 127) + 1);
+            let two_to_127_var5 = self.setup_init_stack((1u128 << 127) + 1, (1u128 << 127) - 1);
+            let two_to_127_var6 = self.setup_init_stack((1u128 << 127) + 1, 1u128 << 127);
+            let two_to_127_var7 = self.setup_init_stack((1u128 << 127) + 1, (1u128 << 127) + 1);
+            let max_plus_max = self.setup_init_stack(u128::MAX, u128::MAX);
+
+            vec![
+                rhs_is_zero,
+                lhs_is_zero,
+                rhs_is_zero_lhs_max,
+                lhs_is_zero_rhs_max,
+                both_are_2_to_127,
+                two_to_127_var0,
+                two_to_127_var1,
+                two_to_127_var2,
+                two_to_127_var3,
+                two_to_127_var4,
+                two_to_127_var5,
+                two_to_127_var6,
+                two_to_127_var7,
+                max_plus_max,
+            ]
+        }
```

Comment:
You'll have better test coverage[^cov], better maintainability, and (in my opinion) better readability if you don't generate the plus/minus one wiggling yourself, but let the machine do it for you.

```rust
fn corner_case_initial_states(&self) -> Vec<Vec<BFieldElement>> {
    let points_with_plus_minus_one = [0, 0x200000002fffffffffff908f8, 1 << 127, u128::MAX]
        .into_iter()
        .flat_map(|p| [p.checked_sub(1), Some(p), p.checked_add(1)])
        .flatten()
        .collect_vec();

    points_with_plus_minus_one
        .iter()
        .cartesian_product(&points_with_plus_minus_one)
        // .filter(|(&l, &r)| l.checked_add(r).is_some()) // discard overflowing pairs?
        .map(|(&l, &r)| self.setup_init_stack(l, r))
        .collect()
}
```

[^cov]: the current test declares 14 cases, my suggestion declares 100